sortdb/check.py

```diff
@@ -0,0 +1,214 @@
+    'db_mtime'
+'''
+
+# stdlib
+import sys
+import os
+import json
+import time
+from urlparse import urlparse
+
+# 3rd party
+import requests
+from requests.exceptions import Timeout, HTTPError, InvalidURL, ConnectionError
+from simplejson import JSONDecodeError
+from hashlib import md5
+
+# project
+from checks import AgentCheck
+
+# Metric types
+GAUGE = 'gauge'
+COUNTER = 'counter'
+
+
+SORTDB_METRICS = {
+    'total_requests' : ('sortdb.stats.total_requests',GAUGE),
+    'total_seeks': ('sortdb.stats.total_seeks',GAUGE),
+    'get_requests': ('sortdb.stats.get_requests',GAUGE),
+    'get_hits': ('sortdb.stats.get_hits',GAUGE),
+    'get_misses': ('sortdb.stats.get_misses',GAUGE),
+    'get_average_request': ('sortdb.stats.get_requests.avg',GAUGE),
+    'get_95': ('sortdb.stats.get_requests.95percentile',GAUGE),
+    'get_99': ('sortdb.stats.get_requests.99percentile',GAUGE),
+    'mget_requests': ('sortdb.stats.mget_requests',GAUGE),
+    'mget_hits': ('sortdb.stats.mget_hits',GAUGE),
+    'mget_misses': ('sortdb.stats.mget_misses',GAUGE),
+    'mget_average_request': ('sortdb.stats.mget_requests.avg',GAUGE),
+    'mget_95': ('sortdb.stats.mget_requests.95percentile',GAUGE),
+    'mget_99': ('sortdb.stats.mget_requests.99percentile',GAUGE),
+    'fwmatch_requests': ('sortdb.stats.fwmatch_requests',GAUGE),
+    'fwmatch_hits': ('sortdb.stats.fwmatch_hits',GAUGE),
+    'fwmatch_misses': ('sortdb.stats.fwmatch_misses',GAUGE),
+    'fwmatch_average_request': ('sortdb.stats.fwmatch_requests.avg',GAUGE),
+    'fwmatch_95': ('sortdb.stats.fwmatch_requests.95percentile',GAUGE),
+    'fwmatch_99': ('sortdb.stats.fwmatch_requests.99percentile',GAUGE),
+    'range_requests': ('sortdb.stats.range_requests',GAUGE),
+    'range_hits': ('sortdb.stats.range_hits',GAUGE),
+    'range_misses': ('sortdb.stats.range_misses',GAUGE),
+    'range_average_request': ('sortdb.stats.range_requests.avg',GAUGE),
+    'range_95': ('sortdb.stats.range_requests.95percentile',GAUGE),
+    'range_99': ('sortdb.stats.range_requests.99percentile',GAUGE),
+    'db_size': ('sortdb.stats.db_size.bytes',GAUGE),
+    'db_mtime': ('sortdb.stats.db_mtime',GAUGE)
+
+}
+
+
+EVENT_TYPE = SOURCE_TYPE_NAME = 'sortdb'
+
+# Service check
+SORTDB_SERVICE_CHECK = 'sortdb.http.can_connect'
+
+class SortdbCheck(AgentCheck):
+    
+
+    def check(self, instance):
+        sortdb_url = instance.get('url')
+        if sortdb_url is None:
+            raise Exception('The Sortdb http api stats url must be specified in the configuration')
+        
+        # get tags for instances to differentiate multiple instances
+        o = urlparse(sortdb_url)
+        instance_tags = []
+ 
+        if all([o.hostname, o.port]):
+            instance_tags = ["sortdb_instance:{0}-{1}".format(o.hostname, o.port)]
+
+        # get and set metrics
+        self._get_sortdb_metrics(sortdb_url, SORTDB_METRICS, instance_tags)
+
+        #service check
+        self.service_check(SORTDB_SERVICE_CHECK,
```

Comment:
I would check health before getting metrics. That way you can reply with a CRITICAL and exit before trying all the metrics endpoints.

Reply:
Good point! will change this.